src/libclipper/include/clipper/task_executor.hpp

```diff
@@ -353,23 +415,38 @@ class TaskExecutor {
   void on_response_recv(rpc::RPCResponse response) {
     std::unique_lock<std::mutex> l(inflight_messages_mutex_);
     auto keys = inflight_messages_[response.first];
+    boost::shared_lock<boost::shared_mutex> metrics_lock(model_metrics_mutex_);
 
     inflight_messages_.erase(response.first);
     std::vector<float> deserialized_outputs =
         deserialize_outputs(response.second);
     assert(deserialized_outputs.size() == keys.size());
     int batch_size = keys.size();
     predictions_counter->increment(batch_size);
-    throughput_meter->mark(batch_size);
+    throughput_meter_->mark(batch_size);
     long current_time = std::chrono::duration_cast<std::chrono::milliseconds>(
```

Comment:
This is a duration in `milliseconds`, but the model prediction latency meter's unit is `microseconds`. This mistake is also present in line 280 of `task_executor_hpp` (milliseconds should be changed to microseconds in the duration cast - I think this is my fault, sorry). In #127, I've modified the `inflight message` entry to include the raw chrono datetime object, rather than a parsed long value - hopefully this helps reduce these issues in the future.